File: GNGradientView/GNGradientView.m
//
//  GNGradientView.m
//  GNGradientView
//
//  Created by Jakub Knejzlik on 21/10/14.
//  Copyright (c) 2014 Jakub Knejzlik. All rights reserved.
//

#import "GNGradientView.h"

@implementation GNGradientView

-(id)initWithCoder:(NSCoder *)aDecoder{
    self = [super initWithCoder:aDecoder];
    if (self) {
        [self setGradientMaskAlphaChannels:@[@1,@0]];
        [self updateGradientMaskFrame];
        [self setGradientMaskDirection:GNGradientViewDirectionBottom];
    }
    return self;
}

@end

File: GNGradientView/Classes/GNGradientView.swift
//
//  GNGradientView.swift
//  Pods
//
//  Created by Jakub Knejzlik on 19/09/16.
//
//

import UIKit

@IBDesignable class GNGradientView: UIView {
    
    @IBInspectable var topAlpha: CGFloat = 1 {
        didSet {
            updateAlphas()
        }
    }
    @IBInspectable var middleAlpha: CGFloat = 0.5 {
        didSet {
            updateAlphas()
        }
    }
    @IBInspectable var bottomAlpha: CGFloat = 0 {
        didSet {
            updateAlphas()
        }
    }
    @IBInspectable var vertical: Bool = true {
        didSet {
            gradientView_setMaskDirection(value: vertical ? .vertical : .horizontal)
        }
    }
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        initialize()
    }
    
    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
        initialize()
    }
    
    func initialize() {
        updateAlphas()
        gradientView_setMaskDirection(value: .vertical)
    }
    
    func updateAlphas() {
        gradientView_setMaskAlphaChannels([topAlpha,middleAlpha,bottomAlpha])
    }
    
    override func layoutSubviews() {
        super.layoutSubviews()
        gradientView_updateMaskFrame()
    }
    
    override func prepareForInterfaceBuilder() {
//        backgroundColor = UIColor.green
    }
}
